Cmake installs icons in system directory with `CREATE_ZIP=on` and `--target bundle`

### What is the bug or the crash?

```ps
cmake --build build --config RelWithDebInfo --target bundle
[1/1] Running CPACK. Please wait...
FAILED: CMakeFiles/bundle D:/QGIS_src/build/CMakeFiles/bundle
C:\Windows\system32\cmd.exe /C "cd /D D:\QGIS_src\build && "C:\Program Files\CMake\bin\cpack.exe" --config D:/QGIS_src/build/BundleConfig.cmake"
CPack: Create package using ZIP
CPack: Install projects
CPack: - Install project: qgis []
Installing crssync ...
Installing crashhandler ...
CMake Error at D:/QGIS_src/build/src/app/cmake_install.cmake:44 (file):
  file cannot create directory: C:/Program Files (x86)/qgis/icons.  Maybe
  need administrative privileges.
Call Stack (most recent call first):
  D:/QGIS_src/build/src/cmake_install.cmake:47 (include)
  D:/QGIS_src/build/cmake_install.cmake:285 (include)
```

config command:

```ps
cmake -S . `
>>   -B build `
>>   -G Ninja `
>>   -D WITH_VCPKG=ON `
>>   -D BUILD_WITH_QT6=ON `
>>   -D WITH_QTWEBKIT=OFF `
>>   -D VCPKG_TARGET_TRIPLET=x64-windows-release `
>>   -D VCPKG_HOST_TRIPLET=x64-windows-release `
>>   -D CMAKE_BUILD_TYPE=RelWithDebInfo `
>>   -D WITH_DESKTOP=ON `
>>   -D WITH_3D=ON `
>>   -D WITH_BINDINGS=ON `
>>   -D ENABLE_TESTS=OFF `
>>   -D USE_CCACHE=ON `
>>   -D CMAKE_C_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
>>   -D CMAKE_CXX_COMPILER_LAUNCHER="C:\ccache\ccache.exe" `
>>   -D VCPKG_INSTALL_OPTIONS="--x-buildtrees-root=C:\vcpkg\src" `
>>   -D FLEX_EXECUTABLE="C:\Flex_bison\win_flex.exe" `
>>   -D BISON_EXECUTABLE="C:\Flex_bison\win_bison.exe" `
>>   -D CREATE_ZIP=ON
```


### Steps to reproduce the issue

.

### Versions

.

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

If I specify `-D CMAKE_INSTALL_PREFIX="D:\Qgis_local"`

```ps
PS D:\QGIS_src> tree /F ..\Qgis_local\
Structure du dossier pour le volume DATA
Le numéro de série du volume est 2244-1DA2
D:\QGIS_LOCAL
└───icons
        qgis-mime.ico
        qgis-qgs.ico
        qgis-qlr.ico
        qgis-qml.ico
        qgis-qpt.ico
        qgis.ico

PS D:\QGIS_src>
```

--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -678,7 +678,7 @@ if (WITH_PDAL)
 endif()
 
 if(MSVC)
-  install(FILES qgis.ico qgis-mime.ico qgis-qgs.ico qgis-qlr.ico qgis-qml.ico qgis-qpt.ico DESTINATION ${CMAKE_INSTALL_PREFIX}/icons)
+  install(FILES qgis.ico qgis-mime.ico qgis-qgs.ico qgis-qlr.ico qgis-qml.ico qgis-qpt.ico DESTINATION icons)
 endif()
 
 install(TARGETS qgis_app

